lib/JSON/Validator/Util.pm

```diff
@@ -18,7 +18,22 @@ our @EXPORT_OK
 sub E { JSON::Validator::Error->new(@_) }
 
 sub data_checksum {
-  Mojo::Util::md5_sum(Data::Dumper->new([@_])->Sortkeys(1)->Useqq(1)->Dump);
+  my $key;
+  if (scalar(@_) == 1 && !(ref $_[0])) {
+    if (!defined $_[0]) {
+      $key = '$VAR1 = undef;';
+    }
+    elsif ($_[0] =~ /^\d+$/) {
+      $key = '$VAR1 = ' . $_[0] . ';';
+    }
+    else {
+      $key = '$VAR1 = "' . $_[0] . '";';
+    }
+  }
+  else {
+    $key = Data::Dumper->new([@_])->Sortkeys(1)->Useqq(1)->Indent(0)->Dump;
+  }
+  Mojo::Util::md5_sum($key);
```

Comment:
I think this can be condensed quite a bit:

```
sub data_checksum {
  return Mojo::Util::md5_sum(
    ref $_[0]
    ? Data::Dumper->new([@_])->Indent(0)->Sortkeys(1)->Terse(1)->Useqq(1)->Dump
    : defined $_[0] ? qq("$_[0]")
    :                 'undef');
}
```

Reply:
Works for me. Want me to do it or would you rather do that yourself since there's not much of my code anymore :) ?

Reply:
It is your code though, just formatted a bit differently 😄It all depends on if you want your name in the code or not I guess - You decide 👍

Reply:
Ok, updated. :) I also had to reintroduce the numeric check, otherwise tests would fail.

Reply:
Oh. That's terrible. In that case, I think you need to use Scalar::Util::looks_like_number(), because your regex fails on `3.14`.

Reply:
I deliberately made it work like Data::Dumper, which would also return 3.14 as `'3.14'`.

Reply:
Can you tell me which test that fail without the numeric test?

Reply:
Uh, err... none. I could swear there was something but I must have made a mistake somewhere. Perhaps I had the "randomly select Data::Dumper instead" debug code there at that time.